XLS-0033d-compact-fungible-tokens/README.md

```diff
@@ -175,16 +185,15 @@ The **`CFToken`** object represents an amount of a token held by an account that
 ##### 1.2.1.2.1 Fields
 A **`CFToken`** object can have the following fields. The key of each CFToken is stored in the Owner Directory for the account that holds the `CFToken`.
 
-| Field Name            | Required?          | JSON Type | Internal Type |
-| --------------------- |--------------------|-----------|---------------|
+| Field Name          | Required?          | JSON Type | Internal Type |
+|---------------------|--------------------|-----------|---------------|
 | `LedgerEntryType`   | :heavy_check_mark: | `number`  | `UINT16`      |
-| `Account`            | :heavy_check_mark: | `string`  | `ACCOUNTID`   |
-| `CFTokenIssuanceID`   | :heavy_check_mark: |  `string` | `UINT256`     |
-| `CFTAmount`              | :heavy_check_mark: |  `string` | `UINT64`      |
-| `LockedAmount`        | default            |  `string` | `UINT64`      |
-| `Flags`               | default            |  `number` | `UINT32`      |
-| `OwnerNode`               | default            |  `number` | `UINT64`      |
-| `CFTokenNode`               | default            |  `number` | `UINT64`      |
```

Comment:
> What is the `CFT Directory` that you're referring to? Are you proposing a new object here?

CFT directory is a directory that stores a list of `CFToken` keys that all share the same `CFTokenIssuance`. This allows the ledger to quickly lookup all the `CFTokens` that all share the same `CFTokenIssuanceID` , and therefore, we can support such API (`cft_holder_balances`) natively on rippled. Also it allows us to implement (if needed) clean-up operations for dangling `CFTokens` (that still live on a ledger after the `CFTokenIssuance` they point to has been deleted)


> Currently, there are only two types of [Owner Directory[(https://xrpl.org/directorynode.html#directorynode) nodes -- one for `offers` and one for `everything else`.
> 
> My understanding from our last discussion was that `CFTokenIssuance` and `CFToken` would both live in their owner's owner dirctories (the "everything else" kind, above).


The CFT directory has exactly the same format as a "owner directory", except it's a directory for a particular `CFTokenIssuance`. CFT directory is using the existing `DirectoryNode` object, but, instead of using the `Owner` field in the object, we introduce a new field `CFTokenIssuanceID` to specify the particular CFT that this directory belongs to. This directory is merely used for bookkeeping on the ledger and is not owned by any account. 

This directory is very important if we want to API natively in rippled and provide flexible options down the roads (see first paragraph). This was something we discussed and I thought we all agreed on this, but maybe I missed something. 

Also would like some insights from @gregtatcam if he has any opinion

Reply:
If you are curious, here is the [PR](https://github.com/shawnxie999/rippled/pull/8/files#diff-a7aebc50b442bcf1c1ab0c6146552459edd8e44b1bae44a67c9035e9cb3b3658R77) implementation.

In `LedgerFormats.cpp`, I added a new optional field to the `DirectoryNode` as 

                               `{sfCFTokenIssuanceID, soeOPTIONAL}`

Reply:
So IMO we should use the CFT directory and keep the `CFTokenNode` on `CFToken` object

Reply:
Just catching up to these comments -- I misunderstood/forgot the purpose of `CFTokenNode`. I'll put the entry back in and ping when it's fixed.

Reply:
Following up here. First, see c2a635cb377bee61ff8188eaa7f228e6ee22d49a where I've added back the `CFTokenNode` to the fields list and table.

That said, (1) I'm still not persuaded that the requirement here is one we should be solving for. (2) Even if we all agreed on the requirement, I'm likewise not persuaded that the design being proposed is the most ideal. 

So, I've added a new section in the `Outstanding Issues` appendix that I think summarizes the overall proposal, and talks about some tradeoffs and alternatives. I tried to capture the rationale you added above in PR comments, as well as some other perspectives (some of them mine). I tried to be neutral, so feel free to suggest edits if you see them.

Reply:
Thanks!

Reply:
Resolving this one to clear the path for a merge.